pygyro/advection/advection.py

```diff
@@ -448,85 +474,84 @@ def step( self, f: np.ndarray, dt: float, phi: Spline2D, v: float ):
             The parallel velocity coordinate
 
         """
-        assert(f.shape==self._nPoints)
-        self._interpolator.compute_interpolant(f,self._spline)
+        assert(f.shape == self._nPoints)
+        self._interpolator.compute_interpolant(f, self._spline)
 
         phiBases = phi.basis
         polBases = self._spline.basis
 
         if (self._explicit):
-            poloidal_advection_step_expl( f, dt, v, self._points[1],
-                            self._points[0], self._drPhi_0,
-                            self._dqPhi_0, self._drPhi_k,
-                            self._dqPhi_k, self._endPts_k1_q,
-                            self._endPts_k1_r, self._endPts_k2_q,
-                            self._endPts_k2_r, phiBases[0].knots,
-                            phiBases[1].knots, phi.coeffs,
-                            phiBases[0].degree, phiBases[1].degree,
-                            polBases[0].knots, polBases[1].knots,
-                            self._spline.coeffs, polBases[0].degree,
-                            polBases[1].degree, self._constants.CN0,
-                            self._constants.kN0, self._constants.deltaRN0,
-                            self._constants.rp, self._constants.CTi,
-                            self._constants.kTi, self._constants.deltaRTi,
-                            self._constants.B0, self._nulEdge)
+            poloidal_advection_step_expl(f, float(dt), v, self._points[1],
```

Comment:
Surely the implicit advection also has this problem? I'm surprised it doesn't appear in other operators. Wouldn't it be safer to cast in `fullSimulation.py`?